### PR TITLE
[ADD] Functional tests

### PR DESCRIPTION
**Pull request by StefanRijnhart**
Continued from https://github.com/OCA/openupgradelib/pull/96 by @bodi000 

* Fixes set_defaults in 7.0
* Fix flake8 not being run on the actual code
* Flake8 fixes
* Add a dedicated test module with tests for 6.1 up to 11.0
* Replace python loading test with running an actual migration from 6.1 to 11.0

### CODE REVIEW

**Comment by StefanRijnhart** [comment]
I think I'm getting this travis test scripting all wrong, so if you have any suggestions please come forth.

**Comment by StefanRijnhart** [comment]
In spite of the rough approach, I'm quite pleased with the fact that we can test openupgradelib methods across Odoo versions nw.

**Comment by StefanRijnhart** [comment]
@pedrobaeza @hbrunn still curious what you think. Functional tests for openupgradelib, what's not to like ;-)

File: openupgradelib/openupgrade.py [hunk 1]
```diff
@@ -808,42 +815,38 @@ def write_value(ids, field, value):
                 cr.execute(query, (params[0], sub_ids))
 
     for model in default_spec.keys():
-        obj = pool.get(model, False)
-        if obj == False:
+        if api and isinstance(pool, api.Environment):
+            obj = pool[model]
+        else:
+            try:
```
Comment (hbrunn) [inline openupgradelib/openupgrade.py]: why not this unconditionally?

**Comment by paulius-sladkevicius** [comment]
@StefanRijnhart would be possible to have `Fixes set_defaults in 7.0` separately done, because now it's included in a functional tests that is not directly related and w/o this fix it's hard to migrate, needs to do manual patching.

**Comment by mreficent** [comment]
@StefanRijnhart what do you say of @paulius-sladkevicius comment? I think there are several people in v6.1 and v7 with the set_defaults problem...

**Comment by StefanRijnhart** [comment]
Well, I still have to see any better proposal than this so let me just rebase this and maybe we can merge it. It can always be refactored later.

**Comment by StefanRijnhart** [comment]
Rebase done.

@pedrobaeza please reconsider to merge this until you have put forward your alternative test strategy.

@hbrunn I've ignored your suggestion about different jobs per version so far because this PR was blocked by Pedro anyway but I have to say I do like the fact that the migration runs an upgrade across all supported versions.

**Comment by pedrobaeza** [comment]
OK, let's go with this approach for now until we have a split way and covering all versions.

**Comment by agb80** [comment]
Can we manke a new release for be able to get this patch easily?

**Comment by pedrobaeza** [comment]
Get it with `pip install --ignore-installed git+https://github.com/OCA/openupgradelib.git@master`

**Pull request merged**